pystac_client/client.py

```diff
@@ -236,45 +299,62 @@ def from_dict(
         return result
 
     @lru_cache()
-    def get_collection(self, collection_id: str) -> Optional[Collection]:
+    def get_collection(
+        self, collection_id: str
+    ) -> Optional[Union[Collection, CollectionClient]]:
         """Get a single collection from this Catalog/API
 
         Args:
             collection_id: The Collection ID to get
 
         Returns:
-            CollectionClient: A STAC Collection
+            Union[Collection, CollectionClient]: A STAC Collection
         """
-        if self._supports_collections() and self._stac_io:
-            url = self._get_collections_href(collection_id)
+        collection: Union[Collection, CollectionClient]
+
+        if self.conforms_to(ConformanceClasses.COLLECTIONS) or self.conforms_to(
+            ConformanceClasses.FEATURES
+        ):
+            assert self._stac_io is not None
+
+            url = self._collections_href()
+            url = f"{url.rstrip('/')}/{collection_id}"
             collection = CollectionClient.from_dict(
                 self._stac_io.read_json(url),
                 root=self,
                 modifier=self.modifier,
             )
             call_modifier(self.modifier, collection)
-            return collection
         else:
-            for col in self.get_collections():
-                if col.id == collection_id:
-                    call_modifier(self.modifier, col)
-                    return col
-
-        return None
+            if self.has_conforms_to():
+                warnings.warn(
+                    DOES_NOT_CONFORM_TO("COLLECTIONS or FEATURES"),
+                    category=DoesNotConformTo,
+                    stacklevel=2,
+                )
```

Comment:
```suggestion
                DoesNotConformTo.warn(ConformanceClasses.COLLECTIONS, ConformanceClasses.FEATURES)
```

This may be too much complexity, but since we have a `warnings.warn(..., category=WarningCat, stacklevel=2)` being repeated everywhere, maybe we could hang a classmethod on `PystacClientWarning` that each warning could customize with their own messages. E.g. `FallbackToPystac` could have `FALLBACK_MSG` baked in, so we don't need to import another constant to other modules.